common/src/main/java/org/apache/gravitino/credential/CredentialProvider.java

```diff
@@ -45,12 +49,24 @@ public interface CredentialProvider extends Closeable {
   String credentialType();
 
   /**
-   * Obtains a credential based on the provided context information.
+   * Gets a credential based on the provided context information.
    *
    * @param context A context object providing necessary information for retrieving credentials.
    * @return A Credential object containing the authentication information needed to access a system
    *     or resource. Null will be returned if no credential is available.
    */
   @Nullable
   Credential getCredential(CredentialContext context);
+
+  /**
+   * Returns the fully qualified class name of the {@link CredentialGenerator} implementation. This
+   * generator will be loaded via reflection to perform the actual credential creation.
+   *
+   * @return The class name of the credential generator.
+   * @throws UnsupportedOperationException if the provider does not use a generator.
+   */
+  default String getGeneratorClassName() {
```

Comment:
any reason to place this class to `CredentialProvider` not `CredentialProviderDelegator `?